### PR TITLE
Map legend

### PR DESCRIPTION
**Pull request by jbalzer12**
Although there is no issue for this topic, I remembered us talking about it. Cause it's no big deal with the help of the used package, I implemented it. If we just forgot about this issue later on and not decided to refuse this idea, you can check this out.

Bildschirm­foto 2023-01-16 um 14 00 28

### CODE REVIEW

**Comment by phuef** [comment]
The legend itself looks good :)
But I would only show the elements when they are in the map

**Comment by jbalzer12** [comment]
> The legend itself looks good :)
> But I would only show the elements when they are in the map

I wasn't able to set this setting :/

**Comment by jbalzer12** [comment]
I now made the bus stations optional. It seems to be not possible to change the this.mapLegend.options.legends changeable as an objects. And if a want to check every possible case it's a lot of code. So I implemented it this way. But I'm still not sure if it's really possible to make it changing depending on the currently shown layers

**Comment by jbalzer12** [comment]
Now the legend is fully responsive :) but still except from the result layer. I'm not sure if that's required

**Comment by phuef** [comment]
Nice! I don't think the result layer is neccessary. 

Do you know if it there is a setting in the legend package that allows the user to hide the legend in a way? Right now the legend takes up quite some space in the mobile view (s. image)
image

**Comment by jbalzer12** [comment]
Bildschirm­foto 2023-01-18 um 23 19 28

it is now visible only by hovering over it. but also in the full screen view

**Review by phuef — APPROVED** [review]
Top 👍  I think it would be optimal, if the legend wouldn't be hidden in the full screen view, but for now this is sufficient for me

File: frontend/src/components/MapView.vue [hunk 1]
```diff
@@ -275,6 +295,59 @@ export default {
...
+        weight: 3,
+      };
+      const busStations = {
+        label: "Bus stations",
+        type: "image",
+        url: busMarker,
+        weight: 2,
+      };
+
+      try {
```
Comment (kardwen) [inline frontend/src/components/MapView.vue:323]: Checking if it exists before removing would be more elegant

**Review by kardwen — APPROVED** [review]

**Pull request merged**